iac-extensions/docker/src/main/java/org/sonar/iac/docker/checks/SecretsHandlingCheck.java

```diff
@@ -125,21 +157,31 @@ private static boolean containsSecretEntityWordCombination(List<String> words) {
     return false;
   }
 
-  private static boolean isSensitiveSecret(@Nullable Argument secret) {
-    return secret != null && isSensitiveValue(ArgumentResolution.of(secret));
-  }
-
-  private static boolean isSensitiveValue(ArgumentResolution valueResolution) {
-    if(valueResolution.is(UNRESOLVED)) return false;
-    String value = valueResolution.value();
-    return !value.isBlank() && !isUrl(value) && !isPath(value);
-  }
-
   private static boolean isUrl(String value) {
     return URL_PATTERN.matcher(value).find();
   }
 
   private static boolean isPath(String value) {
     return PATH_PATTERN.matcher(value).find();
   }
+
+  private static class VarNameSplitter {
+    private static List<String> split(String name) {
+      if (UNDERSCORE_NAME_PATTERN.matcher(name).matches() && name.contains("_")) {
+        return toUpperCase(name.split("_"));
+      }
+      if (DASH_NAME_PATTERN.matcher(name).matches() && name.contains("-")) {
+        return toUpperCase(name.split("-"));
+      }
+      if (CAMELCASE_NAME_PATTERN.matcher(name).matches()) {
+        return toUpperCase(CAMELCASE_SPLIT_PATTERN.split(name));
+      }
+      return Collections.emptyList();
+    }
+
+    private static List<String> toUpperCase(String[] strings) {
+      return Stream.of(strings).map(s -> s.toUpperCase(Locale.ROOT)).collect(Collectors.toList());
+    }
```

Comment:
Possible suggestion : an alternative could be to call `toUpperCase()` on String name before working on it.
The cons is that even if we don't match anything, we will do an operation which costs may not be negligible.
The pros is we simplify a bit the code and don't do streams/collect, which may also be costly, but only happened for matched result.
I don't have a strong opinion on this, I let you decide.